airflow/models/taskinstance.py

```diff
@@ -1245,21 +1245,20 @@ def overwrite_params_with_dag_run_conf(self, params, dag_run):
         if dag_run and dag_run.conf:
             params.update(dag_run.conf)
 
-    def render_templates(self, context=None):
-        task = self.task
+    def render_templates(self, context=None) -> None:
+        """Render templates in the operator fields."""
         if not context:
             context = self.get_template_context()
 
-        if hasattr(self, 'task') and hasattr(self.task, 'dag'):
-            if self.task.dag.user_defined_macros:
-                context.update(self.task.dag.user_defined_macros)
+        if self.task.dag.user_defined_macros:
+            context.update(self.task.dag.user_defined_macros)
 
-        rt = self.task.render_template  # shortcut to method
-        for attr in task.__class__.template_fields:
-            content = getattr(task, attr)
+        # Jinja template all fields in task.template_fields
+        for attr in self.task.template_fields:
+            content = getattr(self.task, attr)
             if content:
-                rendered_content = rt(attr, content, context)
-                setattr(task, attr, rendered_content)
+                rendered_content = self.task.render_template(content, context)
+                setattr(self.task, attr, rendered_content)
```

Comment:
While we are refactoring this: How about we move all this logic in to a function in BaseOperator -- it all seems to operate on `self.task.X` which is probably a good indicator that's where this belongs. (We should keep this for back-compat though?)

Reply:
Good point, will move around some code tonight.